vyper/codegen/core.py

```diff
@@ -107,61 +107,71 @@ def _dynarray_make_setter(dst, src):
         store_length = IRnode.from_list(store_length, annotation=ann)
         ret.append(store_length)
 
+        ret.extend(copy_dynarray_body(dst, src))
+        return ret
+
+    with src.cache_when_complex("darray_src") as (b1, src):
```

Comment:
i think the logic of this branch can pretty much be coalesced with the previous branch now